testsuite_overview.md

```diff
@@ -0,0 +1,310 @@
+specific language implementation of the Rest.li client. As a user, you must first add support for the test suite in your
+desired language. That is why we refer to a general "test suite", and to a "Java TestNG test suite" 
+or a "Python pytest test suite". The general test suite contains the shared data and guidelines, and the language-specific
+test suites use the data and guidelines for testing in a single language.
+
+
+#### How to Run and Add Tests
+For new test suite users, take a look at the [Test Suite: How To](testsuite_how_to.md). This will walk you through running
+the provided Java TestNG suite. It will also explain how to add new test cases to the language-independent test suite.
+
+#### How to Add a New Language
+If you would like to add support for a new language, refer to the [Test Suite New Language Guide](testsuite_new_language.md). This is
+a more involved process, and it will be helpful to go over the Design and Code Structure section of this document first.
+
+
+Design and Code Structure
+----
+The suite is composed of:
+
+* Files in various language independent formats such as .json, .pdsc and .restspec.json.
+* A manifest.json file containing a listing of all the test files to help drive automated test execution.
+* Guidelines on how to use the test suite to validate a rest.li client implementation.
+* A Java TestNG suite that uses this test data to validate the Java Rest.li client implementation.
+
+
+The test suite is structured as follows: 
+```
+.
+|—— client-testsuite
+|   |—— manifest.json
+|   |—— data/
+|   |—— requests/
+|   |—— requests-v2/
+|   |—— responses/
+|   |—— responses-v2/
+|   |—— restspecs/
+|   |—— schemas/
+|   |—— snapshots/
+|   |—— src/
+|   |—— ...
+|—— restli-testsuite-server
+|   |—— src
+|   |—— ...
+|—— build.gradle
+|... 
+```
+### client-testsuite
+The client-testsuite directory contains testing data and the language-specific testing suites, such as the Java TestNG
+suite. 
+
+##### manifest.json
+This manifest file provides machine readable information about all the automated tests provided in this suite.
+It is shared across languages, and does not need to be changed when adding support for a new language. 
+It should be changed when adding new test cases.
+
+This file is intended to help drive portions of the test suite execution. Although it does help, writing additional
+assertions by hand will still be needed in most languages to validate correctness of in-memory representations and language
+bindings.
+
+The file is broken down into a few main sections:
+
+* jsonTestData - list of JSON data files.
+* schemaTestData - list of Rest.li data schema files (.pdsc) as well as JSON data files matching the schemas.
+* wireProtocolTestData - list of Rest.li interface definition files (.restspec.json) as well as test HTTP requests and responses,
+  in the form of files, for operations supported by the interface definition.
+
+##### Language-independent data folders
```

Comment:
How about we standardize a term on calling this? what about "rest.li test suite specification"? And organize this section into
- rest.li TestSuite specification
- Java TestNG Tests
- Sample rest.li server